pkg/controller/common/containerconfig.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"fmt"
+
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	"github.com/openshift/machine-config-operator/pkg/version"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func GenerateContainerConfig(baseMC *mcfgv1.MachineConfig, cc *mcfgv1.ControllerConfig) (*mcfgv1.MachineConfig, error) {
+	// Create deep copy to avoid mutating input
```

Comment:
question: Are still making a DeepCopy() here? If not, let's remove this comment.